File: Solution/CSharp/122.BestTimetoBuyandSellStockII.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int sum = 0;
        for (int tail = prices.Length - 1; tail >= 0;) {
            int head = tail - 1;
            while (head > 0 && prices[head] < prices[head + 1] && prices[head] > prices[head - 1]) {
                head--;
            }
            if (head >= 0) {
                int diff = prices[tail] - prices[head];
                if (diff > 0) {
                    sum += diff;
                }
            }
            tail = head;
        }
        return sum;
    }
}

File: Solution/Python/122. Best Time to Buy and Sell Stock II.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        return sum(max(prices[i + 1] - prices[i], 0) for i in range(len(prices) - 1))


prices = [7, 1, 5, 3, 6, 4]
res = Solution().maxProfit(prices)
print(res)
